src/palace/manager/celery/tasks/reaper.py

```diff
@@ -0,0 +1,239 @@
+from datetime import timedelta
+
+from celery import shared_task
+from sqlalchemy import and_, delete, func, select
+from sqlalchemy.orm import Session, defer, lazyload
+from sqlalchemy.sql import Delete
+from sqlalchemy.sql.elements import or_
+
+from palace.manager.celery.task import Task
+from palace.manager.service.analytics.eventdata import AnalyticsEventData
+from palace.manager.service.celery.celery import QueueNames
+from palace.manager.sqlalchemy.model.circulationevent import CirculationEvent
+from palace.manager.sqlalchemy.model.collection import Collection
+from palace.manager.sqlalchemy.model.credential import Credential
+from palace.manager.sqlalchemy.model.licensing import LicensePool
+from palace.manager.sqlalchemy.model.measurement import Measurement
+from palace.manager.sqlalchemy.model.patron import Annotation, Hold, Loan, Patron
+from palace.manager.sqlalchemy.model.work import Work
+from palace.manager.util.datetime_helpers import utc_now
+from palace.manager.util.log import pluralize
+
+
+def _execute_delete(session: Session, deletion_query: Delete) -> int:
+    # The transactions in the reaper tasks are short lived, so we don't bother to do the
+    # extra work of synchronizing the session with the database after the delete. Once
+    # the transaction is committed, the session contents won't matter anyway.
+    # If we ever need to do something more complex with the session after the delete,
+    # we can revisit this.
+    # https://docs.sqlalchemy.org/en/20/orm/queryguide/dml.html#selecting-a-synchronization-strategy
+    result = session.execute(
+        deletion_query, execution_options={"synchronize_session": False}
+    )
+    # We need the type ignores here because result doesn't always have
+    # a rowcount, but the sqlalchemy docs swear it will in the case of
+    # a delete statement.
+    # https://docs.sqlalchemy.org/en/20/tutorial/data_update.html#getting-affected-row-count-from-update-delete
+    return result.rowcount  # type: ignore[attr-defined,no-any-return]
+
+
+@shared_task(queue=QueueNames.default, bind=True)
+def credential_reaper(task: Task) -> None:
+    """
+    Remove Credentials that expired more than a day ago.
+    """
+    cutoff = utc_now() - timedelta(days=1)
+    deletion_query = delete(Credential).where(Credential.expires < cutoff)
+    with task.transaction() as session:
+        rows_removed = _execute_delete(session, deletion_query)
+    task.log.info(f"Deleted {pluralize(rows_removed, 'expired credential')}.")
```

Comment:
For as many of the reapers as I could, i updated them to use a direct SQL delete statement, rather then loading each record into the ORM, to just delete it. Should allow us to perform the deletes much faster.